scalding-parquet-scrooge/src/test/scala/com/twitter/scalding/parquet/scrooge/ParquetCollectionFormatForwardCompatibilityTests.scala

```diff
@@ -0,0 +1,919 @@
+package com.twitter.scalding.parquet.scrooge
+
+import java.util
+
+import org.apache.parquet.schema.MessageTypeParser
+import org.apache.parquet.thrift.{DecodingSchemaMismatchException, ThriftSchemaConverter}
+import org.apache.parquet.thrift.struct.ThriftField.Requirement
+import org.apache.parquet.thrift.struct.{ThriftField, ThriftType}
+import org.apache.parquet.thrift.struct.ThriftType.StructType.StructOrUnionType
+import org.apache.parquet.thrift.struct.ThriftType.{ListType, MapType, StructType}
+import org.scalatest.{Matchers, WordSpec}
+
+class ParquetCollectionFormatForwardCompatibilityTests extends WordSpec with Matchers {
```

Comment:
ooc how hard would it be to add some actual thrift fixtures based tests in here?

Reply:
ack. Will see how I can do end-to-end testing in this repo.